fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/servlet/PushNotificationsInitializer.java

```diff
@@ -0,0 +1,84 @@
+package pt.ist.fenixedu.integration.servlet;
+
+import java.util.Locale;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+
+import org.fenixedu.bennu.core.rest.JsonBodyReaderWriter;
+import org.fenixedu.cms.domain.Category;
+import org.fenixedu.cms.domain.Post;
+import org.fenixedu.commons.stream.StreamUtils;
+import org.fenixedu.learning.domain.executionCourse.ExecutionCourseSite;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import pt.ist.fenixedu.integration.FenixEduIstIntegrationConfiguration;
+import pt.ist.fenixframework.dml.runtime.RelationAdapter;
+
+@WebListener
+public class PushNotificationsInitializer implements ServletContextListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(PushNotificationsInitializer.class);
+
+    private static final Client HTTP_CLIENT = ClientBuilder.newBuilder().register(JsonBodyReaderWriter.class).build();
+
+    private final static String pushNotificationServerUrl =
+            FenixEduIstIntegrationConfiguration.getConfiguration().getPushNotificationsServer();
+    private final static String header = FenixEduIstIntegrationConfiguration.getConfiguration().getPushNotificationsHeader();
+    private final static String token = FenixEduIstIntegrationConfiguration.getConfiguration().getPushNotificationsToken();
+
+    @Override
+    public void contextInitialized(ServletContextEvent event) {
+        Post.getRelationCategoryPost().addListener(new RelationAdapter<Post, Category>() {
+
+            @Override
+            public void afterAdd(Post post, Category category) {
+                super.afterAdd(post, category);
```

Comment:
@nurv we need Post signals in fenixedu-cms leading to a new minor, how feasible is that in the short term ?
 It looks like you did it already : https://github.com/FenixEdu/fenixedu-cms/pull/16 ?